tools/circt-lec/LogicExporter.cpp

```diff
@@ -0,0 +1,391 @@
+    LLVM_DEBUG(lec::dbgs << "Visiting " #MLIR_NAME "\n");                      \
+    INDENT();                                                                  \
+    LLVM_DEBUG(debugOperands(op));                                             \
+    mlir::Value result = op.getResult();                                       \
+    LLVM_DEBUG(debugOpResult(result));                                         \
+    circuit->perform##OP_NAME(result, op.getOperands());                       \
+    return mlir::success();                                                    \
+  }
+
+// This macro implements the visiting function for a `comb` operation accepting
+// two operands.
+#define visitBinaryCombOp(OP_NAME, MLIR_NAME, TYPE)                            \
+  mlir::LogicalResult LogicExporter::Visitor::visitComb(                       \
+      TYPE op, Solver::Circuit *circuit) {                                     \
+    LLVM_DEBUG(lec::dbgs << "Visiting " #MLIR_NAME "\n");                      \
+    INDENT();                                                                  \
+    LLVM_DEBUG(debugOperands(op));                                             \
+    mlir::Value lhs = op.getLhs();                                             \
+    mlir::Value rhs = op.getRhs();                                             \
+    mlir::Value result = op.getResult();                                       \
+    LLVM_DEBUG(debugOpResult(result));                                         \
+    circuit->perform##OP_NAME(result, lhs, rhs);                               \
+    return mlir::success();                                                    \
+  }
+
+// This macro implements the visiting function for a `comb` operation accepting
+// one operand.
+#define visitUnaryCombOp(OP_NAME, MLIR_NAME, TYPE)                             \
+  mlir::LogicalResult LogicExporter::Visitor::visitComb(                       \
+      TYPE op, Solver::Circuit *circuit) {                                     \
+    LLVM_DEBUG(lec::dbgs << "Visiting " #MLIR_NAME "\n");                      \
+    INDENT();                                                                  \
+    LLVM_DEBUG(debugOperands(op));                                             \
+    mlir::Value input = op.getInput();                                         \
+    mlir::Value result = op.getResult();                                       \
+    LLVM_DEBUG(debugOpResult(result));                                         \
+    circuit->perform##OP_NAME(result, input);                                  \
+    return mlir::success();                                                    \
+  }
+
+visitVariadicCombOp(Add, comb.add, circt::comb::AddOp &);
+
+visitVariadicCombOp(And, comb.and, circt::comb::AndOp &);
+
+visitVariadicCombOp(Concat, comb.concat, circt::comb::ConcatOp &);
+
+visitBinaryCombOp(DivS, comb.divs, circt::comb::DivSOp &);
+
+visitBinaryCombOp(DivU, comb.divu, circt::comb::DivUOp &);
```

Comment:
The same macro issues appear here as in `Circuit.cpp`

```suggestion
visitVariadicCombOp(Add, comb.add, circt::comb::AddOp &)

visitVariadicCombOp(And, comb.and, circt::comb::AndOp &)

visitVariadicCombOp(Concat, comb.concat, circt::comb::ConcatOp &)

visitBinaryCombOp(DivS, comb.divs, circt::comb::DivSOp &)

visitBinaryCombOp(DivU, comb.divu, circt::comb::DivUOp &)
```